hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -1233,15 +1234,41 @@ public boolean isEmpty() {
 
     @Override
     public void putAll(Map<? extends K, ? extends V> m) {
-        Map<Data, Data> map = new HashMap<Data, Data>();
+        ClientPartitionService partitionService = getContext().getPartitionService();
+        Map<Integer, Map<Data, Data>> entryMap = new HashMap<Integer, Map<Data, Data>>(partitionService.getPartitionCount());
+        
         for (Entry<? extends K, ? extends V> entry : m.entrySet()) {
+            checkNotNull(entry.getKey(), NULL_KEY_IS_NOT_ALLOWED);
+            checkNotNull(entry.getValue(), NULL_VALUE_IS_NOT_ALLOWED);
+            
             final Data keyData = toData(entry.getKey());
             invalidateNearCache(keyData);
-            map.put(keyData, toData(entry.getValue()));
+            
+            int partitionId = partitionService.getPartitionId(keyData);
+            if (!entryMap.containsKey(partitionId)) {
+                entryMap.put(partitionId, new HashMap<Data, Data>());
+            }
+            
+            entryMap.get(partitionId).put(keyData, toData(entry.getValue()));
+        }
+        
```

Comment:
The code above this line is trying to classify all the entries matching to the same partition ID but do we really need this information? We are not doing any batch send to the member, we are still sending each request one by one. Hence, we do we need the entryMap at all? Just iterate through m, find the partition id for the key and send it right away.